org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -1289,31 +1289,112 @@ class FileEvent {
 /**
  * Value-object describing what options formatting should use.
  */
-@JsonRpcData
-class FormattingOptions {
+class FormattingOptions extends LinkedHashMap<String, Either<String, Either<Number, Boolean>>> {
```

Comment:
but in the protocol is js number, for custom properties float, double, etc. should be supported